installer/pkg/config-generator/generator.go

```diff
@@ -518,3 +524,21 @@ func tectonicCloudProvider(platform config.Platform) string {
 	}
 	panic("invalid platform")
 }
+
+// Returns a libvirt URI given for the machine-api-operator given the
+// cluster network in CIDR format.
+func libvirtURI(networkCIDR string) (string, error) {
+	_, ipNet, err := net.ParseCIDR(networkCIDR)
+	if err != nil {
+		return "", err
+	}
+
+	gateway, err := cidr.Host(ipNet, 1)
+	if err != nil {
+		return "", err
+	}
+
+	uri := fmt.Sprintf("qemu+tls://%s/system?pkipath=%s", gateway.String(), libvirtPKIPath)
```

Comment:
Ah, this is why you can drop the IP from the config.  I'm a bit nervous about assuming QEMU.  Won't that break non-Linux libvirt even more (#201).  Can we stick with the IP-form in the config and drop the `libvirtURI` helper?

Reply:
Hmm, I'd really like to stick with something close to this. The current default in the config is kind of a hack. It's just relying on a popularly configured default network that is totally unrelated to the infrastructure being defined for the cluster.

What if I parse the URL that's in the config here and generate this a little more carefully, keeping the driver and other options?

Reply:
Changed this to be a bit smarter and added some tests.